animation.go

```diff
@@ -53,16 +64,50 @@ func NewAnimation(d time.Duration, fn func(float32)) *Animation {
 	return &Animation{Duration: d, Tick: fn}
 }
 
+// NewIndefiniteAnimation creates an indefinite animation where the callback function will be called
+// for every rendered frame once started, until stopped.
+//
+// Since: 2.6
+func NewIndefiniteAnimation(fn func()) *IndefiniteAnimation {
+	return &IndefiniteAnimation{Tick: fn}
+}
+
 // Start registers the animation with the application run-loop and starts its execution.
 func (a *Animation) Start() {
 	CurrentApp().Driver().StartAnimation(a)
 }
 
+// Start registers the animation with the application run-loop and starts its execution.
+func (i *IndefiniteAnimation) Start() {
+	i.setupAnimation()
+	i.animation.Start()
+}
+
 // Stop will end this animation and remove it from the run-loop.
 func (a *Animation) Stop() {
 	CurrentApp().Driver().StopAnimation(a)
 }
```

Comment:
I'm not a fan of how the methods of different types are interleaved. I'd much rather have all methods for one typ after each other in alphabetical order before the next type.

Reply:
To clarify: I'd rather have struct, methods, struct, methods and so on. I think that's how Go files generally are structured. Without your latest change it is struct, struct, methods, methods?